tests/integration/Owncloud/OcisPhpSdk/DriveTest.php

```diff
@@ -575,4 +575,228 @@ public function testGetMyDriveForDriveShareReceiver(): void
+        // ocis stable doesn't support root endpoint
+        if (getenv('OCIS_VERSION') === "stable") {
+            $this->markTestSkipped("This method is not implemented in this ocis version");
+        }
+        $shareRoles = $this->drive->getRoles();
+        foreach ($shareRoles as $role) {
+            if ($role->getId() === self::getPermissionsRoleIdByName('Space Viewer')) {
+                $viewerRole = $role;
+                break;
+            }
+        }
+
+        if (empty($viewerRole)) {
+            throw new \Error(
+                "Space viewer role not found "
+            );
+        }
+        try {
+            $this->drive->invite($marie, $viewerRole);
+            $marieReceivedProjectDrive = $marieOcis->getDriveById($this->drive->getId());
+            $this->expectException(ForbiddenException::class);
+            $this->expectExceptionMessage("accessDenied - add grant: error: permission denied:");
+            $marieReceivedProjectDrive->invite($katherine, $viewerRole);
+
+        } catch(EndPointNotImplementedException) {
+            // test should fail if ocis version is less than 6.0.0
+            $this->fail("EndPointNotImplementedException was thrown unexpectedly");
+        }
+    }
+
+    /**
+     * @throws ForbiddenException
+     * @throws InvalidResponseException
+     * @throws BadRequestException
+     * @throws EndPointNotImplementedException
+     * @throws UnauthorizedException
+     * @throws HttpException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     * @throws \Exception
+     */
+    public function testReceiverUpdatesDriveShareRole(): void
+    {
+        $marieOcis = $this->initUser('marie', 'radioactivity');
+        $marie = $this->ocis->getUsers('marie')[0];
+
+        // ocis stable doesn't support root endpoint
+        if (getenv('OCIS_VERSION') === "stable") {
+            $this->markTestSkipped("This method is not implemented in this ocis version");
+        }
+
+        $shareRoles = $this->drive->getRoles();
+        foreach ($shareRoles as $role) {
+            if ($role->getId() === self::getPermissionsRoleIdByName('Manager')) {
+                $managerRole = $role;
+                break;
+            }
+        }
+
+        if (empty($managerRole)) {
+            throw new \Error(
+                "manager role not found "
+            );
+        }
+
+
+        try {
+            $driveInvitation = $this->drive->invite($marie, $managerRole);
+            $permissionId = $driveInvitation->getId();
+            if (empty($permissionId)) {
+                throw new \Error(" Permission not found of user Marie Curie");
+            }
+            $receivedInvitationDrive = $marieOcis->getDriveById($this->drive->getId());
+
+            $shareRoles = $receivedInvitationDrive->getRoles();
+
+            foreach ($shareRoles as $role) {
+                if ($role->getId() === self::getPermissionsRoleIdByName('Space Viewer')) {
```

Comment:
What will happen if "Space Viewer" is somehow not one of the roles that is found?
I suppose that this loop will end with no assert happening. Probably the test will fail because PHPunit will see that the test did no assert.
But maybe it would be nicer to add some logic to detect that "nothing interesting happened" in this for loop, and fail in that case.
```
$spaceViewerRoleFound = false;
foreach { ...
  if () {
    code
    $spaceViewerRoleFound = true;
  }
}
$this->assertTrue($spaceViewerRoleFound, 'Space Viewer role was not found');
```